File: java/v3.1/src/main/java/com/google/api/services/samples/dfareporting/advertisers/GetAdvertiserLandingPages.java
// Copyright 2017 Google Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package com.google.api.services.samples.dfareporting.advertisers;

import com.google.api.client.util.Strings;
import com.google.api.services.dfareporting.Dfareporting;
import com.google.api.services.dfareporting.model.AdvertiserLandingPagesListResponse;
import com.google.api.services.dfareporting.model.LandingPage;
import com.google.api.services.samples.dfareporting.DfaReportingFactory;

/**
 * This example lists all landing pages for a specified advertiser.
 */
public class GetAdvertiserLandingPages {
  private static final String USER_PROFILE_ID = "ENTER_USER_PROFILE_ID_HERE";

  private static final String ADVERTISER_ID = "ENTER_ADVERTISER_ID_HERE";

  public static void runExample(Dfareporting reporting, long profileId, long advertiserId)
      throws Exception {
    // Limit the fields returned.
    String fields = "nextPageToken,landingPages(id,name,url)";

    AdvertiserLandingPagesListResponse result;
    String nextPageToken = null;

    do {
      // Create and execute the advertiser landing pages list request.
      result = reporting.advertiserLandingPages().list(profileId).setFields(fields)
          .setPageToken(nextPageToken).execute();

      for (LandingPage landingPage : result.getLandingPages()) {
        System.out.printf(
            "Advertiser landing page with ID %d, name \"%s\", and URL %s was found.%n",
            landingPage.getId(), landingPage.getName(), landingPage.getUrl());
      }

      // Update the next page token.
      nextPageToken = result.getNextPageToken();
    } while (!result.getLandingPages().isEmpty() && !Strings.isNullOrEmpty(nextPageToken));
  }

  public static void main(String[] args) throws Exception {
    Dfareporting reporting = DfaReportingFactory.getInstance();

    long profileId = Long.parseLong(USER_PROFILE_ID);
    long advertiserId = Long.parseLong(ADVERTISER_ID);

    runExample(reporting, profileId, advertiserId);
  }
}

File: python/v5/get_advertiser_landing_pages.py
#!/usr/bin/python
#
# Copyright 2017 Google Inc. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""This example lists all landing pages for the specified advertiser."""

import argparse
import sys

import dfareporting_utils
from oauth2client import client

# Declare command-line flags.
argparser = argparse.ArgumentParser(add_help=False)
argparser.add_argument(
    'profile_id',
    type=int,
    help='The ID of the profile to look up advertiser landing pages for')
argparser.add_argument(
    'advertiser_id',
    type=int,
    help='The ID of the advertiser to look up landing pages for')


def main(argv):
  # Retrieve command line arguments.
  flags = dfareporting_utils.get_arguments(argv, __doc__, parents=[argparser])

  # Authenticate and construct service.
  service = dfareporting_utils.setup(flags)

  profile_id = flags.profile_id
  advertiser_id = flags.advertiser_id

  try:
    # Construct the request.
    request = service.advertiserLandingPages().list(
        profileId=str(profile_id), advertiserIds=[advertiser_id]
    )

    while True:
      # Execute request and print response.
      response = request.execute()

      for landing_page in response['landingPages']:
        print('Found advertiser landing page with ID %s and name "%s".' %
              (landing_page['id'], landing_page['name']))

      if response['landingPages'] and response['nextPageToken']:
        request = service.advertiserLandingPages().list_next(request, response)
      else:
        break

  except client.AccessTokenRefreshError:
    print('The credentials have been revoked or expired, please re-run the '
          'application to re-authorize')


if __name__ == '__main__':
  main(sys.argv)
